base/subarray.jl

```diff
@@ -324,6 +324,33 @@ find_extended_inds(::ScalarIndex, I...) = (@_inline_meta; find_extended_inds(I..
 find_extended_inds(i1, I...) = (@_inline_meta; (i1, find_extended_inds(I...)...))
 find_extended_inds() = ()
 
+MemoryLayout(A::SubArray) = submemorylayout(MemoryLayout(parent(A)), parentindices(A))
+submemorylayout(::MemoryLayout{T}, _) where T = UnknownLayout{T}()
+submemorylayout(::DenseColumns{T}, ::Tuple{I}) where {T,I<:Union{AbstractUnitRange{Int},Int,AbstractCartesianIndex}} =
+    DenseColumnMajor{T}()
```

Comment:
Shouldn't this be `DenseColumnsStridedRows`?

Reply:
This should only ever be called when `parent(A)` is a vector, so the current implementation is correct:
```julia
julia> A = rand(5,5)

julia> view(A,:) |>Base.MemoryLayout
Base.DenseColumnMajor{Float64}()

julia> B = view(A,1:2,:);

julia> B |> Base.MemoryLayout
Base.DenseColumnsStridedRows{Float64}()

julia> view(B,:) |>Base.MemoryLayout # the parent is actually `ReshapedArray`
Base.UnknownLayout{Float64}()
```
 But in this case the memory layout is always `DenseColumnMajor`, so I think it's clearer if I rewrite this as 
```julia
submemorylayout(::DenseColumnMajor{T}, ::Tuple{I}) where {T,I<:Union{AbstractUnitRange{Int},Int,AbstractCartesianIndex}} =
    DenseColumnMajor{T}()
```

Reply:
I'm not sure which cases below you were referring to, but the other cases such as 
```julia
submemorylayout(::DenseColumns{T}, ::Tuple{I,Int}) where {T,I<:Union{AbstractUnitRange{Int},Int,AbstractCartesianIndex}} =
    DenseColumnMajor{T}()
```
are correct since if the columns are dense, then taking a view of a single column is `DenseColumnMajor`.